src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let join2String = '';
+
+    if (this.length === 1) {
+      return `${this[0]}`;
+    }
+
+    if (this.length > 1) {
+      join2String = `${this[0]}`;
+
+      if (this[0] === null || this[0] === undefined) {
```

Comment:
just start iterating from `0` index

just add a separator after an element always except the last element